File: Typescript/BestTimeToBuyAndSellStockIII.ts
function maxProfit(prices: number[]): number {

    let firstBuy:number = -100000, secondBuy:number = -100000;
    let firstSell:number = 0, secondSell:number = 0;

    for(let i = 0; i < prices.length;i++){
        firstBuy = Math.max(firstBuy, -prices[i]);
        firstSell = Math.max(firstSell, firstBuy+prices[i]);
        secondBuy = Math.max(secondBuy, firstSell-prices[i]);
        secondSell = Math.max(secondSell, secondBuy+prices[i]);
    }

    return secondSell;
    
};

File: Python/BestTimeToBuyAndSellStockIII.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        First_Buy = float("-inf")
        First_Sell = 0
        Second_Buy = float("-inf")
        Second_Sell = 0

        for price in prices:
            First_Buy = max(First_Buy, -price)
            First_Sell = max(First_Sell, First_Buy + price)

            Second_Buy = max(Second_Buy, First_Sell - price)
            Second_Sell = max(Second_Sell, Second_Buy + price)

        return Second_Sell
